### DISCUSSION TITLE
balancer在连接前会检测该出站是否可用吗?

### DISCUSSION THREAD

**Discussion by Weltolk** [XTLS/Xray-core#3171] [Q&A]
balancer在连接前会检测该出站是否可用吗?

**Comment by vrnobody** [comment]
不会。检测工作是由“burstObservatory”或者“observatory"负责，他们会把检测结果记录下来。leastLoad和leastPing挑节点的时候会使用这些信息。而random和roundRobin则不需要这些信息。

**Reply by vrnobody** [reply]
You need to enable [mertics](https://xtls.github.io/en/config/metrics.html#related-configurations) feature in config.json. And then use web browser to access http://127.0.0.1:11111/debug/vars (modify the port parameter accroding to your setting).

Here is a disscussion related to leastLoad strategy https://github.com/XTLS/Xray-core/discussions/3735.

**Reply by vrnobody** [reply]
Observatory is a very old feature, and burstObservatory is a new feature added later. Now we can just use burstObservatory. The subtle difference between those two is that observatory can do sequential probing, while burstObservatory can only probe in random order.

**Reply by vrnobody** [reply]
I don't know why it doesn't work on your side. But I can get "delay" and "health_ping" information with the following config.
```json
{
  "log": {
    "loglevel": "debug"
  },
  "metrics": {
    "tag": "metrics",
    "listen": "127.0.0.1:1111"
  },
  "burstObservatory": {
    "subjectSelector": [
      "proxy"
    ],
    "pingConfig": {
      "destination": "https://www.google.com/generate_204",
      "connectivity": "",
      "interval": "10s",
      "sampling": 2,
      "timeout": "30s"
    }
  },
  "outbounds": [
    {
      "tag": "proxy1"
      ...
    },
    {
      "tag": "proxy2"
      ...
    }
  ]
}
```

**Reply by vrnobody** [reply]
I didn't dig that deep. You need to read the source code

**Comment by Weltolk** [comment]
> 不会。检测工作是由“burstObservatory”或者“observatory"负责，他们会把检测结果记录下来。leastLoad和leastPing挑节点的时候会使用这些信息。而random和roundRobin则不需要这些信息。

leastLoad和leastPing是balancer的strategy? 文档里貌似没有

https://github.com/XTLS/Xray-core/issues/3164

https://github.com/XTLS/Xray-core/issues/323

目前的需求这个issue里有提, 连接之前检测是否可用

**Comment by vrnobody** [comment]
没多少人喜欢写文档，新功能只能看发布说明：https://github.com/XTLS/Xray-core/releases
我没注意到那两个issue，看来我的信息又落伍了，需要更新了。
